### ISSUE TITLE
rake error

### ISSUE THREAD

**Issue by igroykt** [marius-balteanu/redmine_tags#8]
>RAILS_ENV=production rake db:migrate --trace
(in /opt/wwwservices/redmine/www)
** Invoke db:migrate (first_time)
** Invoke environment (first_time)
** Execute environment
rake aborted!
SyntaxError: /opt/wwwservices/redmine/www/plugins/redmine_tags/lib/redmine_tags/patches/issue_patch.rb:33: syntax error, unexpected tLPAREN_ARG, expecting keyword_do_LAMBDA or tLAMBEG
          scope :on_project, -> (project) {
                                 ^
/opt/wwwservices/redmine/www/plugins/redmine_tags/lib/redmine_tags/patches/issue_patch.rb:33: syntax error, unexpected tLAMBEG, expecting keyword_end
/opt/wwwservices/redmine/www/plugins/redmine_tags/lib/redmine_tags/patches/issue_patch.rb:36: syntax error, unexpected '}', expecting keyword_end
/opt/wwwservices/redmine/www/plugins/redmine_tags/lib/redmine_tags/patches/issue_patch.rb:40: module definition in method body
/opt/wwwservices/redmine/www/plugins/redmine_tags/lib/redmine_tags/patches/issue_patch.rb:132: syntax error, unexpected $end, expecting keyword_end
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:360:in `require_or_load'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:494:in `load_missing_constant'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:184:in `const_missing'
/opt/wwwservices/redmine/www/plugins/redmine_tags/init.rb:44:in `block in <top (required)>'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:444:in `instance_exec'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:444:in `block in make_lambda'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:190:in `call'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:190:in `block in simple'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:502:in `call'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:502:in `block in call'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:502:in `each'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:502:in `call'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:88:in `run_callbacks'
/usr/local/lib/ruby/gems/1.9.1/gems/actionpack-4.2.3/lib/action_dispatch/middleware/reloader.rb:83:in `prepare!'
/usr/local/lib/ruby/gems/1.9.1/gems/actionpack-4.2.3/lib/action_dispatch/middleware/reloader.rb:55:in `prepare!'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/application/finisher.rb:50:in `block in <module:Finisher>'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/initializable.rb:30:in `instance_exec'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/initializable.rb:30:in `run'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/initializable.rb:55:in `block in run_initializers'
/usr/local/lib/ruby/1.9.1/tsort.rb:150:in `block in tsort_each'
/usr/local/lib/ruby/1.9.1/tsort.rb:183:in `block (2 levels) in each_strongly_connected_component'
/usr/local/lib/ruby/1.9.1/tsort.rb:219:in `each_strongly_connected_component_from'
/usr/local/lib/ruby/1.9.1/tsort.rb:182:in `block in each_strongly_connected_component'
/usr/local/lib/ruby/1.9.1/tsort.rb:180:in `each'
/usr/local/lib/ruby/1.9.1/tsort.rb:180:in `each_strongly_connected_component'
/usr/local/lib/ruby/1.9.1/tsort.rb:148:in `tsort_each'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/initializable.rb:54:in `run_initializers'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/application.rb:352:in `initialize!'
/opt/wwwservices/redmine/www/config/environment.rb:14:in `<top (required)>'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
/usr/local/lib/ruby/gems/1.9.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/application.rb:328:in `require_environment!'
/usr/local/lib/ruby/gems/1.9.1/gems/railties-4.2.3/lib/rails/application.rb:457:in `block in run_tasks_blocks'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/usr/local/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:199:in `each'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:199:in `invoke_prerequisites'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/usr/local/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
/usr/local/lib/ruby/gems/1.9.1/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'
Tasks: TOP => db:migrate => environment

Environment:
  Redmine version                3.1.0.stable
  Ruby version                   1.9.3-p551 (2014-11-13) [x86_64-linux]
  Rails version                  4.2.3
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.6.11
  Git                            1.7.1
  Filesystem                     
Redmine plugins:
  redmine_apijs                  5.2.0
  redmine_checklists             3.1.1
  redmine_slack                  0.1

**Comment by igroykt** [comment]
problem was in space between -> and (project).
now i have new problem.
there is now taggins and tags tables:
>Mysql2::Error: Table 'redmine.taggings' doesn't exist: SHOW FULL FIELDS FROM `taggings`
Completed 500 Internal Server Error in 988ms (ActiveRecord: 11.2ms)

>ActiveRecord::StatementInvalid (Mysql2::Error: Table 'redmine.taggings' doesn't exist: SHOW FULL FIELDS FROM `taggings`):
  app/models/query.rb:243:in `build_from_params'
  app/models/issue_query.rb:120:in `build_from_params'
  app/helpers/queries_helper.rb:196:in `retrieve_query'
  app/controllers/issues_controller.rb:47:in `index'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'

**Comment by igroykt** [comment]
my bad!
i forgot about rake db:migrate :) know it work

**Comment by stephenyeargin** [comment]
@marius-balteanu @igroykt I think something was left un-done with this one. The original problem seems to require a code change. :simple_smile:

```
diff --git a/lib/redmine_tags/patches/issue_patch.rb b/lib/redmine_tags/patches/issue_patch.rb
index 24d318d..0594bd9 100644
--- a/lib/redmine_tags/patches/issue_patch.rb
+++ b/lib/redmine_tags/patches/issue_patch.rb
@@ -30,7 +30,7 @@ module RedmineTags
           # searchable_options[:columns] << "#{ ActsAsTaggableOn::Tag.table_name }.name"
           # searchable_options[:preload] << :tags
 
-          scope :on_project, -> (project) {
+          scope :on_project, ->(project) {
               project = project.id if project.is_a? Project
               where "#{ Project.table_name }.id = ?", project
             }
diff --git a/lib/redmine_tags/patches/wiki_page_patch.rb b/lib/redmine_tags/patches/wiki_page_patch.rb
index 5b476b9..a395890 100644
--- a/lib/redmine_tags/patches/wiki_page_patch.rb
+++ b/lib/redmine_tags/patches/wiki_page_patch.rb
@@ -30,7 +30,7 @@ module RedmineTags
           # searchable_options[:columns] << "#{ ActsAsTaggableOn::Tag.table_name }.name"
           # searchable_options[:preload] << :tags
 
-          scope :on_project, -> (project) {
+          scope :on_project, ->(project) {
               project = project.id if project.is_a? Project
               where "#{ Project.table_name }.id = ?", project
             }
```

**Comment by stephenyeargin** [comment]
Fixed in #12
